cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -197,5 +209,51 @@ private Path validateAppConfigRelativePath(RepositoryManager repositoryManager,
     }
     return filePath;
   }
+
+  @Override
+  public RepositoryAppsResponse list(NamespaceId namespace, RepositoryConfig repoConfig) throws
+    AuthenticationConfigException, NotFoundException {
+    try (RepositoryManager repositoryManager = repoManagerFactory.create(namespace, repoConfig)) {
+      String currentCommit = repositoryManager.cloneRemote();
+      Path basePath = repositoryManager.getBasePath();
+
+      if (!Files.exists(basePath)) {
+        throw new NotFoundException(String.format("Missing repository base path %s", basePath));
+      }
+
+      FileFilter configFileFilter = getConfigFileFilter();
+      List<File> configFiles = DirUtils.listFiles(basePath.toFile(), configFileFilter);
+      List<RepositoryApp> responses = new ArrayList<>();
+
+      // TODO CDAP-20420 Implement bulk fetch file hash in RepositoryManager
+      for (File configFile : configFiles) {
+        String applicationName = FileUtils.getNameWithoutExtension(configFile.getName());
+        try {
+          String fileHash = repositoryManager.getFileHash(basePath.relativize(configFile.toPath()), currentCommit);
+          responses.add(new RepositoryApp(applicationName, fileHash));
+        } catch (NotFoundException e) {
+          // ignore file if not found in git tree
+          LOG.warn("Skipping config for application {}: {}", applicationName, e.getMessage());
```

Comment:
log.warn as it is not a expected scenario but should not also create a api error